source/Halibut/Transport/SecureClient.cs

```diff
@@ -68,7 +72,7 @@ public void ExecuteTransaction(ExchangeAction protocolHandler, CancellationToken
                     IConnection connection = null;
                     try
                     {
-                        connection = connectionManager.AcquireConnection(protocolBuilder, new TcpConnectionFactory(clientCertificate, asyncHalibutFeature, halibutTimeoutsAndLimits), ServiceEndpoint, log, cancellationToken);
+                        connection = connectionManager.AcquireConnection(protocolBuilder, new TcpConnectionFactory(clientCertificate, asyncHalibutFeature, halibutTimeoutsAndLimits, streamFactory), ServiceEndpoint, log, cancellationToken);
```

Comment:
Note that the streamFactory is passed to the ctr of the TcpConnectionFactory

Reply:
Can we inject the `TcpConnectionFactory` here, and reduce the parameters given to this class.

Similar justification to the other place I already commented 😁